spec/acceptance/reboot_message.spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'spec_helper_acceptance'
+
+describe 'Reboot Module - Custom Message' do
+
+  def apply_reboot_manifest(agent, reboot_manifest)
+    apply_manifest_on agent, reboot_manifest, {:debug => true} do |result|
+      if fact_on(agent, 'kernel') == 'SunOS'
+        expected_command = /shutdown -y -i 6 -g 60 \"A different message\"/
+      else
+        expected_command = /shutdown -r \+1 \"A different message\"/
```

Comment:
If your test does not require checking the actual command, you can just `/shutdown [-a-z0-9+ ]* \"A different message\"/` and remove the special casing here.